lib/helpers/special_routes.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:auto_route/auto_route.dart';
+import 'package:brunstadtv_app/api/brunstadtv.dart';
+import 'package:brunstadtv_app/helpers/navigation/navigation_utils.dart';
+import 'package:brunstadtv_app/helpers/redirect.dart';
+import 'package:collection/collection.dart';
+import 'package:firebase_crashlytics/firebase_crashlytics.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:url_launcher/url_launcher.dart';
+
+import '../graphql/client.dart';
+
+class SpecialRoutesHandler {
+  static Future<bool> handle(BuildContext context, String path) async {
+    ProviderContainer ref;
+    if (!context.mounted) return false;
+    try {
+      ref = ProviderScope.containerOf(context, listen: false);
+    } catch (e, st) {
+      FirebaseCrashlytics.instance.recordError(e, st);
+      return false;
+    }
+
+    if (!path.startsWith('/')) return false;
+    var uri = Uri.tryParse('https://web.brunstad.tv$path');
+    if (uri == null || uri.pathSegments.isEmpty) return false;
+
+    bool handleTvLogin() {
```

Comment:
Is it better to have nested functions instead of instance methods?

Reply:
It's mostly a taste thing, but as long as this file doesn't get crazy big I think nested like this is a good pragmatic way.